carla-carma-integration/Dockerfile

```diff
@@ -0,0 +1,73 @@
+# Copyright (C) 2021 LEIDOS.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+FROM ubuntu:16.04
+FROM ros:kinetic
+WORKDIR /home
+
+# CARLA PythonAPI
+RUN mkdir ./PythonAPI
+ADD https://carla-releases.s3.eu-west-3.amazonaws.com/Backup/carla-0.9.10-py2.7-linux-x86_64.egg ./PythonAPI
+
+RUN apt-get update && apt-get install -y \
+		git \
+		curl \
+		wget \
+		nano \
+		libpng16-16 \
+		libsdl2-2.0 \
+		software-properties-common
+
+# Update gcc and g++
+RUN add-apt-repository ppa:ubuntu-toolchain-r/test
```

Comment:
I'd recommend moving this apt-add repository (and any others) above the first apt-get update, so that we only need to run apt-get update once. Should save on build time by a good bit.

Reply:
The first apt-get update will install software-properties-common (line 29) which allow the system to use add-apt-repository command. As a result, we might need to run apt-get update twice for this Dockerfile